external/vcm/vcm/safe.py

```diff
@@ -30,7 +31,7 @@ def _validate_stack_dims(ds, dims, allowed_broadcast_dims=()):
 
 
 def stack_once(
-    ds: xr.Dataset,
+    ds: Union[xr.DataArray, xr.Dataset],
```

Comment:
This change may cause a lot of mypy headaches, because the return type is now `Union[xr.DataArray, xr.Dataset]`. Can you use [generic typing](https://docs.python.org/3/library/typing.html#typing.Generic) to say the return type is the type of `ds`? I think you'll specifically want to use `T = TypeVar('T', xr.DataArray, xr.Dataset)`